packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -142,157 +172,137 @@ export class SocketModeClient extends EventEmitter {
-      .on('explicit disconnect')
-        .transitionTo('disconnecting')
-      .on('server disconnect warning')
-        .transitionTo('reconnecting').withCondition(() => this.autoReconnectEnabled)
-      .on('server pings not received')
-        .transitionTo('reconnecting').withCondition(() => this.autoReconnectEnabled)
-      .on('server pongs not received')
-        .transitionTo('reconnecting').withCondition(() => this.autoReconnectEnabled)
-      .on('server disconnect old socket')
-        .transitionTo('closing-socket')
+      .submachine(this.connectedStateMachine)
+      .on(Event.ServerDisconnectWarning)
+        .transitionTo(State.Reconnecting).withCondition(() => this.autoReconnectEnabled)
+      .on(Event.WebSocketClose)
+        .transitionTo(State.Reconnecting).withCondition(() => this.autoReconnectEnabled)
+        .transitionTo(State.Disconnected).withAction(this.handleDisconnection.bind(this))
+      .on(Event.ExplicitDisconnect)
+        .transitionTo(State.Disconnecting)
+      .on(Event.ServerDisconnectWarning)
+        .transitionTo(State.Reconnecting).withCondition(() => this.autoReconnectEnabled)
+      .on(Event.ServerPingsNotReceived)
+        .transitionTo(State.Reconnecting).withCondition(() => this.autoReconnectEnabled)
+      .on(Event.ServerPongsNotReceived)
+        .transitionTo(State.Reconnecting).withCondition(() => this.autoReconnectEnabled)
+      .on(Event.ServerDisconnectOldSocket)
+        .transitionTo(State.ClosingSocket)
       .onExit(() => {
         this.connected = false;
         this.authenticated = false;
-        if (this.serverPingTimeout !== undefined) {
-          clearTimeout(this.serverPingTimeout);
-        }
-        if (this.clientPingTimeout !== undefined) {
-          clearTimeout(this.clientPingTimeout);
-        }
+        this.tearDownHeartBeatJobs();
       })
-    .state('disconnecting')
+    .state(State.Disconnecting)
       .onEnter(() => {
-        // Most of the time, a websocket will exist.
+        // Most of the time, a WebSocket will exist.
         // The only time it does not is when transitioning from connecting,
-        // before the client.start() has finished and the websocket hasn't been set up.
+        // before the client.start() has finished and the WebSocket hasn't been set up.
         if (this.websocket !== undefined) {
           this.websocket.close();
         }
       })
-      .on('websocket close')
-        .transitionTo('disconnected')
+      .on(Event.WebSocketClose)
+        .transitionTo(State.Disconnected)
       .onExit(() => this.tearDownWebSocket())
-    // Reconnecting is just like disconnecting,
-    // except that the websocket should already be closed
-    // before we enter this state, and that the next state should be connecting.
-    .state('reconnecting')
+    .state(State.Reconnecting)
+      .onEnter(() => {
+        this.logger.debug('Reconnecting to Slack ...');
```

Comment:
This looks good but is it essentially a duplicate of the global state machine's generic `onStateEnter` event logger at the very top of this state machine's definition? In fact, after running the simple example you provided, this log does not come up at all...?

Reply:
If you don't see this log, the SocketModeClient code might be a bit older one. I've added this log mainly for debugging. We can remove it if it sounds duplicated with other logs.